src/compression.rs

```diff
@@ -1145,6 +1165,14 @@ fn initial_inventions(
             let right_utility = right_edge_utility(right_edge_key(&group[0]), &egraph);
             let compressive_utility = compressive_utility(left_utility + right_utility, &ztuple, &group, num_paths_to_node, egraph, appzipper_of_node_zid);
             let utility = compressive_utility + other_utility(left_utility + right_utility, cfg);
+
+            // prune finished inventions specific to one single task
+            if !cfg.no_opt_single_task
+                    && group.iter().all(|node| tasks_of_node[&node].len() <= 1)
+                    && group.iter().all(|node| tasks_of_node[&group[0]].iter().next() == tasks_of_node[&node].iter().next()) {
```

Comment:
Changes:
* it's safe to do `[0]` instead of `.iter().next()` here
* on the previous line, do `==1` instead of `<=1`
* somewhere around where you build `tasks_of_node` do an `assert tasks_of_node.values().all(|tasks| !tasks.is_empty())`

Reply:
afaik `[0]` doesn't work because you can't index into HashSets: https://doc.rust-lang.org/std/collections/struct.HashSet.html

If `.iter().next()` isn't idiomatic then using set operations could be an alternative: `tasks_of_node[&group[0]].is_subset(tasks_of_node[&node])` for example. I don't have a preference either way so it's whatever you think is more Rust-like

The other two changes have been implemented: [0fa9639](https://github.com/mlb2251/stitch/pull/84/commits/0fa963988431b65b164208c781ce7f1aff39565e)

Reply:
could also just do set equality: `tasks_of_node[&group[0] == tasks_of_node[&node]`. Same semantically either way since they're all singleton sets

Reply:
Hm. Yes I see, `.iter().next()` is the best of those options, lets go with it.